core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -260,7 +260,12 @@ private[spark] class BlockManager(
   def waitForAsyncReregister(): Unit = {
     val task = asyncReregisterTask
     if (task != null) {
-      Await.ready(task, Duration.Inf)
+      try {
+        Await.ready(task, Duration.Inf)
```

Comment:
According to the Scaladoc (and actual usages), it looks like this particular `waitForAsyncReregister` method is only used in test code and I'm guessing that it's probably called from within an interrupt-based timeout block.

As for the other usages, we'd have to consider them on a case-by-case basis.